// rollup-plugin-obfuscator/dist/rollup-plugin-obfuscator.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var pluginutils_1 = require("@rollup/pluginutils");
var javascript_obfuscator_1 = require("javascript-obfuscator");
var defaultOptions = {
    global: false,
    options: {},
    include: ['**/*.js', '**/*.ts'],
    exclude: ['node_modules/**'],
    obfuscate: javascript_obfuscator_1.obfuscate,
};
function rollupPluginObfuscator(override) {
    var options = __assign(__assign({}, defaultOptions), override);
    var filter = (0, pluginutils_1.createFilter)(options.include, options.exclude);
    return {
        name: 'rollup-plugin-obfuscator',
        transform: options.global ? undefined : function (code, id) {
            if (!filter(id))
                return null;
            var obfuscationResult = options.obfuscate(code, __assign(__assign({}, options.options), { inputFileName: id, sourceMap: true }));
            return {
                code: obfuscationResult.getObfuscatedCode(),
                map: obfuscationResult.getSourceMap(),
            };
        },
        renderChunk: !options.global ? undefined : function (code, _a) {
            var fileName = _a.fileName;
            var obfuscationResult = options.obfuscate(code, __assign(__assign({}, options.options), { inputFileName: fileName, sourceMap: true }));
            return {
                code: obfuscationResult.getObfuscatedCode(),
                map: obfuscationResult.getSourceMap(),
            };
        }
    };
}
exports.default = rollupPluginObfuscator;
// unfortunately, TypeScript won't generate the following for us
// see https://github.com/microsoft/TypeScript/issues/2719
// but we can assume we're always in a CommonJS environment. Right? ...Right?
module.exports = rollupPluginObfuscator;
